core/src/main/scala/kafka/log/OffsetIndex.scala

```diff
@@ -136,7 +136,7 @@ class OffsetIndex(_file: File, baseOffset: Long, maxIndexSize: Int = -1, writabl
 
   /**
    * Append an entry for the given offset/location pair to the index. This entry must have a larger offset than all subsequent entries.
-   * @throws IndexOffsetOverflowException if the offset causes index offset to overflow
+   * @throws InvalidOffsetException if the offset causes index offset to overflow
```

Comment:
This method does indeed throw `IndexOffsetOverflowException` (from `relativeOffset()`) for the reason listed. 

I'm assuming you got confused because it also throws `InvalidOffsetException`. It's ok to add another `@throws` tag if you want but we don't want to remove the existing one.

Reply:
Thanks for taking time to review @mimaison!

I've added the exception kept the original one.